src/compile/common.ts

```diff
@@ -127,6 +131,10 @@ export function numberFormat(fieldDef: TypedFieldDef<string>, specifiedFormat: s
   return undefined;
 }
 
+function getFormatType(fieldDef: FieldDef<string>): 'number' | 'time' {
+  return isPositionFieldDef(fieldDef) && fieldDef.axis ? fieldDef.axis.formatType : undefined;
```

Comment:
What about format in `legend` and in text encoding?

I'd:

1) Check if the fieldDef is a TextFieldDef   (Add a type guard `isTextFieldDef()`, which checks for the containment of `format` or `formatType` )
2)  call the `getGuide()` method and read `formatType` from it.